### DISCUSSION TITLE
How to invoke rules_rust toolchain rustc via the command line?

### DISCUSSION THREAD

**Discussion by samuela** [bazelbuild/rules_rust#2027] [Q&A]
Supposing I have a toolchain set up with eg
```
# WORKSPACE
rust_register_toolchains(
    edition = "2021",
    extra_target_triples = [],
    urls = ["https://foo.com/{}.tar.gz"],
    versions = ["1.66.1"],
)
```
How can I invoke this toolchain's `rustc` via the command line? For example, something like
```
bazel run @rules_rust//tools/rustc -- --version
```

**Comment by UebelAndre** [comment]
There isn't really a fantastic way, you'd need to know what toolchain you matched. There's [@rules_rust//rust/toolchain:current_rustc_files](https://github.com/bazelbuild/rules_rust/blob/0.24.1/rust/toolchain/BUILD.bazel#L16-L19) but that's not executable. It might be reasonable to do so but building that target should force bazel to download a compiler where you can find it in the Bazel output base for your checkout.

**Reply by samuela** [reply]
Is the compiler downloaded to a stable location? Could that potentially be a workaround?
